### PR TITLE
Wildcard [Tooltip]: Support controlled open prop

### PR DESCRIPTION
**Pull request by vovakulikov**
## Background

This PR introduces a new set of props in the Tooltip UI (`open` + `onOpenChange`). These props allow us to have full control over the tooltip's appearance. Sometimes, it may be useful if you want to have custom action triggers to render or not tooltip UI. For example, in the code insights, we're going to have a case when we want to render a tooltip on click only (see [Figma designs](https://www.figma.com/file/aMO6sPoUHDUUcqa25wFnWl/Surface-timeout-limit-hit-to-end-users-in-the-UI-%2339969-Review?node-id=260%3A11)) 

## Test plan
- Check tooltip storybook demos (there should be no regressions) 

## App preview:

- [Web](https://sg-web-vk-support-controlled-tooltip-open.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.02% (+0.51 kb) | 0.01% (+1.02 kb) | 0.00% (+0.51 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/vk/support-controlled-tooltip-open/compare-report.html;tab=live_object) for a full comparison between the commits c171a97 and 6608a62 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/wildcard/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -101,26 +133,33 @@ export const Tooltip: FC<TooltipProps> = props => {
...
+                return
             }
         }
     }
 
-    const isOpen = useDebounce(open, 100)
+    const tooltipId = `tooltip-${useId()}`
+    const isTooltipOpen = useDebounce(isOpen, 100)
+
+    // console.log(isTooltipOpen)
```
Comment (valerybugakov) [inline client/wildcard/src/components/Tooltip/Tooltip.tsx:154]:
```suggestion
```

File: client/wildcard/src/components/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -101,26 +133,33 @@ export const Tooltip: FC<TooltipProps> = props => {
...
-                setOpen(false)
+                setOpen({ isOpen: event.isOpen, reason: TooltipOpenEventReason.ClickOutside })
+                return
             }
         }
     }
 
-    const isOpen = useDebounce(open, 100)
+    const tooltipId = `tooltip-${useId()}`
+    const isTooltipOpen = useDebounce(isOpen, 100)
```
Comment (valerybugakov) [inline client/wildcard/src/components/Tooltip/Tooltip.tsx:152]: To make it clear from the variable name, the difference from `isOpen`.

```suggestion
    const isOpenDebounced = useDebounce(isOpen, 100)
```

File: client/wildcard/src/components/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -1,13 +1,28 @@
...
 import { useId } from '@reach/auto-id'
+import { noop } from 'lodash';
 import { useMergeRefs } from 'use-callback-ref'
 
 import { useDebounce } from '../../hooks'
 import { PopoverContent, PopoverOpenEvent, PopoverOpenEventReason, PopoverTail, Position } from '../Popover'
 
 import styles from './Tooltip.module.scss'
 
+export enum TooltipOpenEventReason {
```
Comment (valerybugakov) [inline client/wildcard/src/components/Tooltip/Tooltip.tsx:12]: At first glance, this name might be confusing. I assumed this enum includes potential reasons for opening a tooltip, but it has also reasons for closing it. Maybe `TooltipVisibilityChangeReason` or `TooltipOpenChangeReason` would be more aligned with the implementation.

**Review by valerybugakov — APPROVED** [review]

**Pull request merged**